pkg/query-service/app/metrics/v4/delta/timeseries.go

```diff
@@ -118,3 +160,37 @@ func prepareMetricQueryDeltaTimeSeries(start, end, step int64, mq *v3.BuilderQue
 
 	return query, nil
 }
+
+// canShortCircuit returns true if we can use the optimized query
+// for the given query
+// This is used to avoid the group by fingerprint thus improving the performance
+// for certain queries
+// cases where we can short circuit:
+// 1. time aggregation = (rate|increase) and space aggregation = sum
```

Comment:
This is `sum_rate` which one of the most used aggregations? What is the perf gained?

Reply:
The last time I measured it makes queries run ~2x faster.

Reply:
nice